packages/light-client/package.json

```diff
@@ -60,7 +60,7 @@
     "build:watch": "yarn run build --watch",
     "build:release": "yarn clean && yarn run build && yarn run build:dist",
     "check-build": "node -e \"(async function() { await import('./lib/index.js') })()\"",
-    "check-types": "tsc",
+    "check-types": "yarn run build:dist && tsc",
```

Comment:
Why do we need this?

Reply:
This checks the tests file as well and one test file is using the types imported from the bundle. So we have to build it before running types check.

Reply:
Can we solve this differently? Having this build run as part of check-types is really annoying during local development. check-types is supposed to be fast and not create a bundle for production

This takes more time than all other check-types across all packages combined
```
@lodestar/light-client: vite v5.2.10 building for production...
@lodestar/light-client: transforming...
@lodestar/light-client: ✓ 562 modules transformed.
@lodestar/light-client: rendering chunks...
@lodestar/light-client: [vite:dts] Start generate declaration files...
@lodestar/light-client: computing gzip size...
@lodestar/light-client: [vite:dts] Start rollup declaration files...
@lodestar/light-client: Analysis will use the bundled TypeScript version 5.4.2
@lodestar/light-client: [vite:dts] Declaration files built in 6702ms.
@lodestar/light-client: dist/manifest.json            0.13 kB │ gzip:   0.11 kB
@lodestar/light-client: dist/lightclient.min.mjs  1,160.16 kB │ gzip: 341.51 kB │ map: 2,887.35 kB
@lodestar/light-client: ✓ built in 12.17s
```

It also creates a untracked file

```
Untracked files:
  (use "git add <file>..." to include in what will be committed)
        packages/light-client/stats.html
```

Reply:
it also randomly fails, see CI https://github.com/ChainSafe/lodestar/actions/runs/8911114728/job/24471848781#step:4:124

```
@lodestar/light-client: test/unit/webEsmBundle.browser.test.ts:3:63 - error TS2307: Cannot find module '../../dist/lightclient.min.mjs' or its corresponding type declarations.
@lodestar/light-client: 3 import {Lightclient, LightclientEvent, utils, transport} from "../../dist/lightclient.min.mjs";
```